core/src/main/java/org/jruby/internal/runtime/methods/MixedModeIRMethod.java

```diff
@@ -85,12 +81,10 @@ public InterpreterContext ensureInstrsReady() {
     public IRubyObject call(ThreadContext context, IRubyObject self, RubyModule clazz, String name, IRubyObject[] args, Block block) {
         if (IRRuntimeHelpers.isDebug()) doDebug();
 
-        DynamicMethodBox box = this.box;
-        if (box.callCount >= 0) tryJit(context, box);
-        DynamicMethod jittedMethod = box.actualMethod;
+        if (callCount >= 0) tryJit(context);
 
-        if (jittedMethod != null) {
-            return jittedMethod.call(context, self, clazz, name, args, block);
+        if (actualMethod != null) {
```

Comment:
used to be one volatile field read, now its 2 right? 
not sure it matters - but I think Java 8 still doesn't re-arrange volatile reads

Reply:
@kares you're right, even though this isn't really contended it's still better to copy. Sec adding the copy back :)

Reply:
Fixed :)